src/monattack.cpp

```diff
@@ -4666,6 +4673,75 @@ bool mattack::riotbot( monster *z )
     return true;
 }
 
+bool mattack::flesh_tendril( monster *z )
+{
+
+    Creature *target = z->attack_target();
+
+    if( target == nullptr || !z->sees( *target ) ) {
+        if( one_in( 70 ) ) {
+            add_msg( _( "The floor trembles underneath your feet." ) );
+            z->moves -= 200;
+            sounds::sound( z->pos(), 60, sounds::sound_t::alert, _( "a deafening roar!" ), false, "shout",
+                           "roar" );
+        }
+        return false;
+    }
+
+    int distance_to_target = rl_dist( z->pos(), target->pos() );
+
+    if( distance_to_target > 3 && one_in( 12 ) ) { // the monster summons stuff to fight you
+
+        std::vector<tripoint> free;
+
+        mtype_id spawned = mon_zombie_gasbag_crawler;
+        if( one_in( 2 ) ) {
+            spawned = mon_zombie_gasbag_impaler;
+        }
+
+        for( const tripoint &dest : g->m.points_in_radius( z->pos(), 1 ) ) {
+            if( g->is_empty( dest ) ) {
+                free.push_back( dest );
+            }
+        }
+        if( !free.empty() ) {
+
+            z->moves -= 100;
+            const tripoint target = random_entry( free );
+            if( monster *const summoned = g->summon_mon( spawned, target ) ) {
+                summoned->make_ally( *z );
+                g->m.propagate_field( z->pos(), fd_gibs_flesh, 75, 1 );
+                if( g->u.sees( *z ) ) {
+                    add_msg( m_warning, _( "A %s struggles to pull itself free from the %s!" ), summoned->name(),
+                             z->name() );
+                }
+            }
+            return true;
+        }
+    }
+
+    if( 2 <= distance_to_target && distance_to_target <= 3 && one_in( 4 ) ) {
```

Comment:
Distance to target is more than or equals to 2 AND distance to target is less than or equals to 3? Will it ever be true?

Reply:
Yes, distance_to_target equal to 2 or 3.

Reply:
Then why don't write it like `distance_to_target == 2 || distance_to_target  == 3`?